### PR TITLE
Add debug/toolbar button to reload changed classes

### LINKED ISSUE #559: [HCR] Action button to apply code changes to running app
The UX of HCR can be improved by making it explicit. With the latest update of VS Code, you can now add a button in the debug toolbar to expose this feature. And this is how Dart guys did:
image

Please refer to this issue for more details: Microsoft/vscode#69153

### PR DESCRIPTION
**Pull request by testforstephen**
Signed-off-by: Jinbo Wang <jinbwan@microsoft.com>
Close #559

Add a new debug settings `java.debug.settings.hotCodeReplace` to control how to trigger HCR. Defaults to `manual`.
-  manual - Click the toolbar to apply the changes.
image
- auto - Automatically apply the changes after compilation. This is the old behavior.
- never - Never apply the changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,7 +95,10 @@ Please also check the documentation of [Language Support for Java by Red Hat](ht
...
 - `java.debug.settings.showLogicalStructure`: show the logical structure for the Collection and Map classes in "Variables" viewlet, defaults to `true`.
 - `java.debug.settings.maxStringLength`: the maximum length of string displayed in "Variables" or "Debug Console" viewlet, the string longer than this length will be trimmed, defaults to `0` which means no trim is performed.
-- `java.debug.settings.enableHotCodeReplace`: enable hot code replace for Java code. Make sure the auto build is not disabled for [VSCode Java](https://github.com/redhat-developer/vscode-java). See the [wiki page](https://github.com/Microsoft/vscode-java-debug/wiki/Hot-Code-Replace) for more information about usages and limitations.
+- `java.debug.settings.hotCodeReplace`: Reload the changed Java classes during debugging, defaults to `manual`. Make sure the auto build is not disabled for [VSCode Java](https://github.com/redhat-developer/vscode-java). See the [wiki page](https://github.com/Microsoft/vscode-java-debug/wiki/Hot-Code-Replace) for more information about usages and limitations.
```
Comment (yaohaizh) [inline README.md]: Just list the setting full name here.

File: src/extension.ts [hunk 2]
```diff
@@ -49,6 +48,29 @@ function initializeExtension(operationId: string, context: vscode.ExtensionConte
...
         return specifyProgramArguments(context);
     }));
+    context.subscriptions.push(instrumentAndRegisterCommand("java.debug.hotCodeReplace", async (args: any) => {
+        const autobuildConfig: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration("java.autobuild");
+        if (!autobuildConfig.enabled) {
+            const ans = await vscode.window.showWarningMessage(
+                "The hot code replace feature requires you to enable the autobuild flag, do you want to enable it?",
+                "Yes", "No");
+            if (ans === "Yes") {
+                await autobuildConfig.update("enabled", true);
```
Comment (yaohaizh) [inline src/extension.ts]: What happen after enable it?

Comment (testforstephen) [reply src/extension.ts]: current hcr doesn't take effect, you need click again.

File: README.md (hunk 1 above)
Comment (testforstephen) [reply README.md]: fixed.

File: package.json [hunk 3]
```diff
@@ -14,7 +14,7 @@
     "debugger"
   ],
   "engines": {
-    "vscode": "^1.22.0"
+    "vscode": "^1.32.0"
```
Comment (yaohaizh) [inline package.json]: When is the features introduced in vscode for additional icons in debug toolbar?

Comment (testforstephen) [reply package.json]: yes, 1.32 add the support to extend the debug toolbar buttons.

**Pull request merged**